x-pack/plugins/reporting/public/share_context_menu/register_csv_reporting.tsx

```diff
@@ -48,26 +48,15 @@ export const csvReportingProvider = ({
       ? moment.tz.guess()
       : uiSettings.get('dateFormat:tz');
 
-  const getShareMenuItems = ({
-    objectType,
-    objectId,
-    sharingData,
-    isDirty,
-    onClose,
-  }: ShareContext) => {
-    if ('search' !== objectType) {
+  const getShareMenuItems = ({ sharingData, onClose }: ShareContext) => {
+    const { taskType, taskConfig } = sharingData;
+    if (taskType !== 'csv') {
       return [];
     }
 
     const jobParams: JobParamsCSV = {
       browserTimezone,
-      objectType,
-      title: sharingData.title as string,
-      indexPatternId: sharingData.indexPatternId as string,
-      searchRequest: sharingData.searchRequest as SearchRequest,
-      fields: sharingData.fields as string[],
-      metaFields: sharingData.metaFields as string[],
-      conflictedTypesFields: sharingData.conflictedTypesFields as string[],
```

Comment:
This changes the object defining the job parameters, which is a breaking change. The job parameters are preserved in "POST URL" strings, which users copy into Watcher for automatic report generation.

Is it possible to keep BWC with the legacy parameters? If not, then we will create a new `ExportTypeDefinition` in Reporting.

Reply:
We'll create a new Export type definition